src/login/LoginOptionsScreen.tsx

```diff
@@ -0,0 +1,227 @@
+import {useAuthState} from '@atb/auth';
+import ThemeText from '@atb/components/text';
+import {VippsSignInErrorCode} from '@atb/auth/AuthContext';
+import {RouteProp, useNavigation} from '@react-navigation/native';
+import {StaticColorByType} from '@atb/theme/colors';
+import {
+  getOrCreateVippsUserCustomToken,
+  authorizeUser,
+} from '@atb/api/vipps-login/api';
+import {useAppStateStatus} from '@atb/utils/use-app-state-status';
+import {parseUrl} from 'query-string';
+import storage from '@atb/storage';
+import {LoginInAppStackParams} from '@atb/login/in-app/LoginInAppStack';
+import {AfterLoginParams} from '@atb/login/types';
+import {VippsLoginButton} from '@atb/components/vipps-login-button';
+import MessageBox from '@atb/components/message-box';
+import * as Sections from '@atb/components/sections';
+
+const themeColor: StaticColorByType<'background'> = 'background_accent_0';
+
+export type LoginOptionsRouteParams = {
+  afterLogin: AfterLoginParams;
+};
+
+type LoginOptionsRouteProps = RouteProp<
+  LoginInAppStackParams,
+  'LoginOptionsScreen'
+>;
+
+type LoginOptionsProps = {
+  route: LoginOptionsRouteProps;
+};
+
+export default function LoginOptionsScreen({
+  route: {
+    params: {afterLogin},
+  },
+}: LoginOptionsProps) {
+  const {t} = useTranslation();
+  const styles = useThemeStyles();
+  const {signInWithCustomToken} = useAuthState();
+  const [error, setError] = useState<VippsSignInErrorCode>();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigation = useNavigation();
+  const appStatus = useAppStateStatus();
+  const [authorizationCode, setAuthorizationCode] = useState('');
+
+  const authenticateUserByVipps = async () => {
+    setIsLoading(true);
+    try {
+      await authorizeUser(setIsLoading);
+    } catch (err) {
+      setError('unknown_error');
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(
+    () =>
+      navigation.addListener('focus', () => {
+        setIsLoading(false);
+      }),
+    [navigation],
+  );
+
+  const signInUsingCustomToken = async (token: string) => {
+    const response = await signInWithCustomToken(token);
+    if (!response.error) {
+      navigation.navigate('Profile');
+    } else {
+      setError(response.error);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const signInVippsUser = async () => {
+      try {
+        const customToken = await getOrCreateVippsUserCustomToken(
+          authorizationCode,
+        );
+        await signInUsingCustomToken(customToken.data as string);
+        await storage.set('vipps_state', '');
+        await storage.set('vipps_nonce', '');
+      } catch (err) {
+        setError('unknown_error');
+        setIsLoading(false);
+      }
+    };
+    if (authorizationCode) {
+      signInVippsUser();
+    }
+  }, [authorizationCode]);
+
+  useEffect(() => {
+    const vippsCallbackHandler = async (event: any) => {
+      if (event.url.includes('atb://auth/vipps')) {
```

Comment:
This should be updated according to this: https://github.com/AtB-AS/mittatb-app/pull/2845
```suggestion
      if (event.url.includes(`${APP_SCHEMA}://auth/vipps`)) {
```

Reply:
Also `VIPPS_CALLBACK_URL` in the BFF

Reply:
@torfjor  Could you please set the env variable accordingly? Might be changes needed at vipps end as well?

Reply:
think` atb-staging://` and `atb:// `should be enough